src/Functions/FunctionsStringDistance.cpp

```diff
@@ -108,6 +108,39 @@ struct ByteHammingDistanceImpl
     }
 };
 
+struct ByteJaccardIndexImpl
+{
+    using ResultType = Float64;
+    static ResultType inline process(
+        const char * __restrict haystack, size_t haystack_size, const char * __restrict needle, size_t needle_size)
+    {
+        if (haystack_size == 0 || needle_size == 0)
+            return 0;
+
+        constexpr size_t max_size = std::numeric_limits<unsigned char>::max() + 1;
+        std::array<UInt8, max_size> haystack_set;
+        std::array<UInt8, max_size> needle_set;
+
+        haystack_set.fill(0);
+        needle_set.fill(0);
+
+        for (size_t i = 0; i < haystack_size; ++i)
+            haystack_set[static_cast<unsigned char>(haystack[i])] = 1;
+        for (size_t i = 0; i < needle_size; ++i)
+            needle_set[static_cast<unsigned char>(needle[i])] = 1;
```

Comment:
Maybe we can combine filling the second array with iteration on the second string?

Reply:
Do you mean filling with zeros?

Or something like this?

```cpp
size_t i = 0;
size_t n = std::min(haystack_size, needle_size);
for (; i < n; ++i)
{
    haystack_set[static_cast<unsigned char>(haystack[i])] = 1;
    needle_set[static_cast<unsigned char>(needle[i])] = 1;
}
for (; i < haystack_size; ++i)
    haystack_set[static_cast<unsigned char>(haystack[i])] = 1;
for (; i < needle_size; ++i)
    needle_set[static_cast<unsigned char>(needle[i])] = 1;

```